PocketKit/Tests/PocketKitTests/Refresh/HomeRefreshCoordinatorTests.swift

```diff
@@ -0,0 +1,135 @@
+        source.stubFetchSlateLineup { _ in }
+        
+        let coordinator = subject()
+        coordinator.refresh {
+            XCTFail("Should not have refreshed")
+        }
+        XCTAssertEqual(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey) as? Date, date)
+    }
+
+    func test_coordinator_whenDataIsStale_refreshesHome() {
+        let date = Calendar.current.date(byAdding: .hour, value: -2, to: Date())
+        userDefaults.setValue(date, forKey: HomeRefreshCoordinator.dateLastRefreshKey)
+        XCTAssertNotNil(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey))
+        source.stubFetchSlateLineup { _ in }
+
+        let coordinator = subject()
+
+        let expectRefresh = expectation(description: "Refresh home")
+        coordinator.refresh {
+            expectRefresh.fulfill()
+        }
+        wait(for: [expectRefresh], timeout: 1)
+        XCTAssertNotEqual(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey) as? Date, date)
+    }
+
+    func test_coordinator_whenForceRefresh_refreshesHome() {
+        XCTAssertNil(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey))
+        source.stubFetchSlateLineup { _ in }
+        let expectRefresh = expectation(description: "Refresh home")
+
+        let coordinator = subject()
+
+        coordinator.refresh(isForced: true) {
+            expectRefresh.fulfill()
+        }
+        wait(for: [expectRefresh], timeout: 1)
+    }
+
+    func test_refresh_delegatesToSource() {
+        let fetchExpectation = expectation(description: "expected to fetch slate lineup")
+        source.stubFetchSlateLineup { _ in fetchExpectation.fulfill() }
+
+        let coordinator = subject()
+
+        coordinator.refresh(isForced: true) { }
+        wait(for: [fetchExpectation], timeout: 2)
+
+        XCTAssertEqual(source.fetchSlateLineupCall(at: 0)?.identifier, "e39bc22a-6b70-4ed2-8247-4b3f1a516bd1")
+    }
+
+    func test_coordinator_whenEnterForeground_whenDataIsNotStale_doesNotRefreshHome() {
+
+        source.stubFetchSlateLineup { _ in
+            XCTFail("Should not fetch slate lineup")
+        }
+        let date = Date()
+        userDefaults.setValue(date, forKey: HomeRefreshCoordinator.dateLastRefreshKey)
+
+        let coordinator = subject()
+        notificationCenter.post(name: UIScene.willEnterForegroundNotification, object: nil)
+
+        XCTAssertEqual(userDefaults.object(forKey: HomeRefreshCoordinator.dateLastRefreshKey) as? Date, date)
+    }
+
+    func test_coordinator_whenEnterForeground_whenDataIsStale_refreshesHome() {
+        let fetchExpectation = expectation(description: "expected to fetch slate lineup")
+        source.stubFetchSlateLineup { _ in fetchExpectation.fulfill() }
+        let date = Calendar.current.date(byAdding: .hour, value: -2, to: Date())
+        userDefaults.setValue(date, forKey: HomeRefreshCoordinator.dateLastRefreshKey)
+
+        let coordinator = subject()
```

Comment:
this gives me a warning message because `coordinator` is not being used, but using `_ = subject()` breaks the test and does not trigger the post, anyone encounter something similar?

Reply:
I wonder if this is some sort of object lifecycle thing. Having an explicit variable definitely keeps it around for the life of this test, so I'm wondering if `_` removes the object from memory immediately (thus, nothing to receive the notification).

Reply:
Using the memory debugger, it confirms that it isn't kept around, so you're right. Wish Xcode didn't leave the warning for it though.
![image](https://user-images.githubusercontent.com/6743397/200334067-797b1d28-be74-4cd7-8da6-04e18dd0ebf7.png)

```
Initialization of immutable value 'coordinator' was never used; consider replacing with assignment to '_' or removing it
```